compiler-core/src/format/tests.rs

```diff
@@ -1123,6 +1123,89 @@ fn expr_int() {
 "#
     );
 
+    assert_format_rewrite!(
```

Comment:
These assertions are very hard to read. Where're the rewrites here? Could we break it down into multiple assertions please. There should also only be one assertion per test.

Reply:
Do we need to test all the number rewrites anyway, it seems we test the same thing again and again? Or would it be suffcient to a few of the rewrites instead of the whole pattern?

Reply:
I don't know what you mean. Tests should cover the changed functionality as appropriate

Reply:
Let me know if its good now or if you just want one line `let #(1_234, _) = #(1_234, Nil)` to `let #(1234, _) = #(1234, Nil)` instead of a subset of the int underscore formatter that I picked now.